portal/src/graphql/portal/AnonymousUsersConfigurationScreen.tsx

```diff
@@ -1,29 +1,122 @@
-import React, { useContext, useCallback } from "react";
+import React, { useContext, useCallback, useState, useMemo } from "react";
 import { useParams } from "react-router-dom";
 import { produce } from "immer";
 import { FormattedMessage, Context } from "@oursky/react-messageformat";
-import { Text, Toggle, Dropdown, PrimaryButton } from "@fluentui/react";
+import {
+  Text,
+  Toggle,
+  Dropdown,
+  PrimaryButton,
+  IDropdownOption,
+} from "@fluentui/react";
 
 import { useAppConfigQuery } from "./query/appConfigQuery";
+import {
+  PortalAPIAppConfig,
+  PromotionConflictBehaviour,
+  promotionConflictBehaviours,
+  isPromotionConflictBehaviour,
+} from "../../types";
 import ShowLoading from "../../ShowLoading";
 import ShowError from "../../ShowError";
 
 import styles from "./AnonymousUsersConfigurationScreen.module.scss";
 
+interface AnonymousUsersConfigurationScreenState {
+  enabled: boolean;
+  promotionConflictBehaviour?: PromotionConflictBehaviour;
+}
+
+function constructStateFromAppConfig(
+  appConfig: PortalAPIAppConfig | null
+): AnonymousUsersConfigurationScreenState {
+  if (appConfig == null) {
+    return {
+      enabled: false,
+    };
+  }
+  const promotionConflictBehaviour = appConfig.identity?.on_conflict?.promotion;
+  return {
+    enabled: promotionConflictBehaviour != null,
+    promotionConflictBehaviour,
+  };
+}
+
+function constructNewAppConfigFromState(
```

Comment:
Similarly, if the toggle is enabled, ensure `appConfig.authentication.identities` contains `anonymous`. Otherwise `appConfig.authentication.identities` must NOT contain `anonymous`.

If the toggle is enabled, the on conflict option is saved. Otherwise, the original value should be used (i.e. keep the original value, do not touch it / delete it)